Mathlib/Combinatorics/Additive/VerySmallDoubling.lean

```diff
@@ -0,0 +1,48 @@
+/-
+Copyright (c) 2024 Yaël Dillies, Patrick Luo. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies, Patrick Luo
+-/
+import Mathlib.Algebra.Group.Pointwise.Finset.Basic
+import Mathlib.GroupTheory.GroupAction.Defs
+
+/-!
+# Sets with very small doubling
+
+This file characterises sets with no doubling (finsets `A` such that `#(A ^ 2) = #A`) as the sets
```

Comment:
Only because you use the word "characterises" here, it would be nice to have the converse

Reply:
The converse is already proved by material from the pointwise API: `∅ ^ 2 = ∅`, `(a • H) ^ 2 = a ^ 2 • H ^ 2 = a ^ 2 • H` (at least once you merge #19259 😇). I don't think it's really worth writing the converse explicitly?

Reply:
Agreed, but let's give a reference to those in that case. I still think it'd be kinda nice to have something of the form `#(A * A) ≤ #A` iff ... but I'm not too bothered